### PR TITLE
[Feature] add RequireFeatureService Annotation

### PR DESCRIPTION
**Pull request by stbischof**
Add RequireFeatureService and FeatureConstants.

How is this `* @author $Id: $` generated?

### CODE REVIEW

File: org.osgi.service.feature/bnd.bnd [hunk 1]
```diff
@@ -4,4 +4,5 @@
 Export-Package: ${p}.*; -split-package:=first
 
 -buildpath: \
-    osgi.annotation;maven-scope=provided;version=${osgi.annotation.version}
+	osgi.annotation;maven-scope=provided;version='${osgi.annotation.version}',\
+	org.osgi.namespace.implementation
```
Comment (bjhargrave) [inline org.osgi.service.feature/bnd.bnd:8]: You must specify a version 1.0 on here like is done in other bnd files.

File: org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0 
+ *******************************************************************************/
+
+package org.osgi.service.feature;
+
+/**
+ * Defines standard constants for the Feature services.
+ * 
+ * @author $Id: $
```
Comment (bjhargrave) [inline org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java:24]: The id should be generated by git during checkout but the string needs to be `$Id$`.

File: org.osgi.service.feature/src/org/osgi/service/feature/annotation/package-info.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * Feature Annotations Package Version 1.0.
+ * <p>
+ * This package contains annotations that can be used to require the
+ * Feature-Service implementation.
+ * <p>
+ * Bundles should not normally need to import this package as the annotations
+ * are only used at build-time.
+ * <p>
+ * 
+ * @author $Id: $
```
Comment (bjhargrave) [inline org.osgi.service.feature/src/org/osgi/service/feature/annotation/package-info.java:29]: `$Id$`

File: org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ */
+public final class FeatureConstants {
+	private FeatureConstants() {
+		// non-instantiable
+	}
+
+	/**
+	 * The name of the implementation capability for the Features specification
+	 */
+	public static final String	FEATURE_IMPLEMENTATION			= "osgi.features";
```
Comment (bjhargrave) [inline org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java:34]: Is it 'feature' or 'features'. I think it should be 'feature' to match the package name, etc.

File: org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ * 
+ * @author $Id: $
+ */
+public final class FeatureConstants {
+	private FeatureConstants() {
+		// non-instantiable
+	}
+
+	/**
+	 * The name of the implementation capability for the Features specification
```
Comment (bjhargrave) [inline org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java:32]: Feature specification

File: org.osgi.service.feature/bnd.bnd (hunk 1 above)
Comment (stbischof) [reply org.osgi.service.feature/bnd.bnd:8]: fixed

File: org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java (hunk 5 above)
Comment (stbischof) [reply org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java:32]: used singular

File: org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java (hunk 4 above)
Comment (stbischof) [reply org.osgi.service.feature/src/org/osgi/service/feature/FeatureConstants.java:34]: used singular

**Comment by stbischof** [comment]
I changed Features to Feature in some more files. Especially the docbook.

@bosschaert Is that okay for you?
I have in my mind that we talked about before, but did not remember the result and couldn't find the discussion.

**Comment by bosschaert** [comment]
> @bosschaert Is that okay for you?
> I have in my mind that we talked about before, but did not remember the result and couldn't find the discussion.

If this is the general consensus, I'm fine with it.

**Comment by stbischof** [comment]
i just squashed the commits together.

**Review by bjhargrave — APPROVED** [review]

**Comment by stbischof** [comment]
rebased to resolve the conflicting file `osgi.specs/docbook/159/service.features.xml`

**Comment by bjhargrave** [comment]
Thanks!

**Pull request merged**